### PR TITLE
Feature: Add ktlint

### PR DESCRIPTION
**Pull request by akueisara**
- [x] Remove usages of JCenter (Ref: https://developer.android.com/studio/build/jcenter-migration)
- [x] Add [ktlint](https://github.com/pinterest/ktlint) 
- [x] Set up a pre-push [githook](https://githooks.com/) for ktlint linting and formatting
- [x] Add `githooks.gradle` to run a task to copy `/.githooks/pre-push` to `.git/hooks`
**Note:** I wanted to keep the pre-push script in the repo, but any files in the .git folder could not be added to version control, so I put the script in the /.githooks folder and created a gradle task 'installGitHooks' in githooks.gradle to add this script to .git/hooks
- [x] Format the code based on the lint check report
- [x] Remove the `kotlin-android-extensions` plugin and use `viewBinding` (Ref: https://developer.android.com/topic/libraries/view-binding/migration)

### CODE REVIEW

File: libsource/build.gradle [hunk 1]
```diff
@@ -47,7 +57,26 @@ dependencies {
...
+task ktlint(type: JavaExec, group: "verification") {
+    description = "Check Kotlin code style."
+    classpath = configurations.ktlint
+    main = "com.pinterest.ktlint.Main"
+    args "src/**/*.kt",
+            // to generate report in checkstyle format prepend following args:
+            "--reporter=plain", "--reporter=html,output=${buildDir}/ktlint.html"
+    // to add a baseline to check against prepend following args:
+    // "--baseline=ktlint-baseline.xml"
+    // see https://github.com/pinterest/ktlint#usage for more
```
Comment (gpark-vs) [inline libsource/build.gradle:73]: Needed to remove some uncommented lines?

**Review by gpark-vs — APPROVED** [review]
Hey Kuei, I have checked PR!

And, as minor feedback, it would be better to split one PR into two steps PR between the reformatted changes and other changes.
It was a bit difficult to recognize what changes are different from reformatted ones.

Overall, it looks great!
I have left one comment for misc things!
Please check it out!

File: libsource/build.gradle (hunk 1 above)
Comment (akueisara) [reply libsource/build.gradle]: sounds good, they are removed.

**Comment by akueisara** [comment]
> Hey Kuei, I have checked PR!
> 
> And, as minor feedback, it would be better to split one PR into two steps PR between the reformatted changes and other changes.
> It was a bit difficult to recognize what changes are different from reformatted ones.
> 
> Overall, it looks great!
> I have left one comment for misc things!
> Please check it out!

@gpark-vs Thanks for the feedback! I formatted the code when I was running some gradle tasks I was not very familiar with in order to pass the lint checks. It was kind of the learning process, so I didn't plan it well to split the changes into different branches.
Next time I'll do my best to separate them or at least leave some comments to let you know what changes are not part of formatting. Thank you!

**Comment by gpark-vs** [comment]
I have checked the changes!
Please feel free to merge when you are okay!

**Pull request merged**